### PR TITLE
Add "snackbar" or notification popups

### PR DESCRIPTION
**Pull request by PaddyEriksen**
This adds a success/failure popup in the lower left corner for:

- Saving layout
- Creating/saving a new widget

E.g.

image
image

Issues:
[FCC-63](https://dotmodus.atlassian.net/browse/FCC-63)
[FCC-64](https://dotmodus.atlassian.net/browse/FCC-64)

### CODE REVIEW

File: Frontend/src/components/DashboardPage/SaveLayoutPrompt.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+const mapDispatchToProps = (dispatch) => ({
+  saveLayout: () => dispatch(saveLayout()),
+  dismissSaveLayout: () => dispatch(dismissSaveLayout()),
+});
+
+SaveLayoutPrompt = withStyles(styles)(SaveLayoutPrompt);
+SaveLayoutPrompt = connect(mapStateToProps, mapDispatchToProps)(SaveLayoutPrompt);
+
+export default SaveLayoutPrompt
```
Comment (DannyLee12) [inline Frontend/src/components/DashboardPage/SaveLayoutPrompt.js]: Missing newline

File: Frontend/src/components/common/SnackBar/SnackBar.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      </div>
+    );
+  }
+}
+
+SimpleSnackbar.propTypes = {
+  classes: PropTypes.object.isRequired,
+};
+
+export default withStyles(styles)(SimpleSnackbar);
```
Comment (DannyLee12) [inline Frontend/src/components/common/SnackBar/SnackBar.js]: Missing newline?

File: Frontend/src/reducers/notifications.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    case HIDE_NOTIFICATION: {
+      return {
+        ...state,
+        showAlert: false,
+      }
+    }
+
+  }
+  return state;
+}
```
Comment (DannyLee12) [inline Frontend/src/reducers/notifications.js]: Missing newline?

**Comment by PaddyEriksen** [comment]
@DannyLee12 ready for rechecking

**Pull request merged**